pkg/dataconn/client.go

```diff
@@ -33,6 +30,8 @@ type Client struct {
 	wire      *Wire
 	peerAddr  string
 	err       error
+	timer     *time.Timer
```

Comment:
I feel we have a race condition here without a lock?

Also call this one `timeoutTimer` instead of `timer`.

Reply:
I renamed it.  The timer is only used by the loop goroutine.